tests/stdio/test_rename.c

```diff
@@ -142,7 +148,24 @@ void test() {
   err = rename("dir", "dir/somename");
   assert(err == -1);
   assert(errno == EINVAL);
-  
+
+  // After changing the current working directory
+  // source should still not be an ancestor of target.
+  // TODO: WasmFS needs to change directory locking to
+  // fix this bug where changing cwd permits this illegal operation.
```

Comment:
```suggestion
  // TODO: WasmFS needs to traverse up the directory tree from the target to detect this case. To do that safely, we will need to globally lock the directory structure to prevent concurrent modifications during the traversal.
```